src/analytical_solutions/standard_models.jl

```diff
@@ -1,59 +1,208 @@
-
-struct OneCompartmentParallelModel <: ExplicitModel end
-function (::OneCompartmentParallelModel)(t,t0,amounts,doses,p,rates)
-  ka1 = p.Ka1
-  ka2 = p.Ka2
-  CL = p.CL
-  V = p.V
-  ke = CL/V         # elimination rate
-  amt = amounts + doses  # initial
-  Sa1 = exp(-(t-t0)*ka1)
-  Sa2 = exp(-(t-t0)*ka2)
-  Se = exp(-(t-t0)*ke)
-
-  Depot1  = amt[1] * Sa1 + rates[1]/ka1*(1-Sa1)          # next depot1 (cmt==1)
-
-  Depot2  = amt[2] * Sa2 + rates[2]/ka2*(1-Sa2)          # next depot2 (cmt==2)
-
-  Central =  ka1 / (ka1 - ke) * (amt[1] * (Se - Sa1) + rates[1]*((1-Se)/ke - (1-Sa1)/ka1)) +
-  ka2 / (ka2 - ke) * (amt[2] * (Se - Sa2) + rates[2]*((1-Se)/ke - (1-Sa2)/ka2)) +
-  amt[3] * Se + rates[3]/ke*(1-Se) # next central (cmt==3)
-  OneCompartmentParallelVector(Depot1,Depot2,Central)
-end
-pk_init(::OneCompartmentParallelModel) = SLVector(Depot1=0.0,Depot2=0.0,Central=0.0)
-
-varnames(::Type{OneCompartmentParallelModel}) = [:Depot1, :Depot2, :Central]
+  T = typeof(Ke)
+
+  Λ = @SVector([-Ke])
+  𝕍 = @SMatrix([T(1)])
+
+  return Λ, 𝕍
+end
+varnames(::Type{OneCmtModel}) = (:Central,)
+pk_init(::OneCmtModel) = SLVector(Central=0.0)
+
+struct OneCmtDepotModel <: ExplicitModel end
+(m::OneCmtDepotModel)(args...) = _analytical_solve(m, args...)
+@inline function LinearAlgebra.eigen(::OneCmtDepotModel, p)
+    a = p.Ka
+    e = p.CL/p.V
+
+    Λ = @SVector([-a, -e])
+    v = e/a - 1
+    𝕍 = @SMatrix([v 0;
+                  1 1])
+
+    return Λ, 𝕍
+end
+varnames(::Type{OneCmtDepotModel}) = (:Depot, :Central)
+pk_init(::OneCmtDepotModel) = SLVector(Depot=0.0,Central=0.0)
+
+struct OneCmtTwoDepotModel <: ExplicitModel end
+(m::OneCmtTwoDepotModel)(args...) = _analytical_solve(m, args...)
+@inline function LinearAlgebra.eigen(::OneCmtTwoDepotModel, p)
+    a = p.Ka1
+    b = p.Ka2
+    e = p.CL/p.V
+
+    frac1 = (e-a)/a
+    invfrac1 = inv(frac1)
+
+    frac2 = (e-b)/b
+    invfrac2 = inv(frac2)
+
+    Λ = @SVector([-a, -b, -e])
+
+    v1 = -1 + e/a
+    v2 = -1 + e/b
+    𝕍 = @SMatrix([frac1 0     0;
+                  0     frac2 0;
+                  1     1     1])
+
+    return Λ, 𝕍
+end
+varnames(::Type{OneCmtTwoDepotModel}) = (:Depot1, :Depot2, :Central)
+pk_init(::OneCmtTwoDepotModel) = SLVector(Depot1=0.0,Depot2=0.0,Central=0.0)
+
+function _Λ(::TwoCmtPeriModel, a, b, c)
+  A = a + b + c
+  S = sqrt(A^2-4*a*c)
+  Λ = @SVector([-(A+S)/2, -(A-S)/2])
+end
+struct TwoCmtPeriModel <: ExplicitModel end
+(m::TwoCmtPeriModel)(args...) = _analytical_solve(m, args...)
+@inline function LinearAlgebra.eigen(m::TwoCmtPeriModel, p)
+    a = p.CL/p.Vc
+    b = p.Q/p.Vc
+    c = p.Q/p.Vp
+
+    Λ = _Λ(m, a, b, c)
+    𝕍 = @SMatrix([(Λ[1]+c)/b (Λ[2]+c)/b;
+                   1          1        ])
+
+    return Λ, 𝕍
+end
+varnames(::Type{TwoCmtPeriModel}) = (:Central, :Peripheral)
+pk_init(::TwoCmtPeriModel) = SLVector(Central=0.0, Peripheral=0.0)
+
+struct TwoCmtDepotPeriModel <: ExplicitModel end
```

Comment:
```suggestion
struct TwoCmtDepotModel <: ExplicitModel end
```

Same note as above - to be consistent with `OneCmt`